quarkus.http.test-ssl-enabled to true does not have an effect in integration test

**Describe the bug**
[(Describe the problem clearly and concisely.)](https://quarkus.io/guides/getting-started-testing#executing-against-a-running-application) 

` quarkus.http.test-ssl-enabled to true` does not have an effect for me with Quarkus 3.21.4. Consider the test class (excerpt) executed with `./mvnw  test -DskipITs=false -Dtest='elexisweb.*IT' -Dsurefire.failIfNoSpecifiedTests=false -Dquarkus.http.test-ssl-enabled=true -Dquarkus.http.test-host=somehost.info -Dquarkus.http.test-port=443` it still resolves to `http:`

```
@QuarkusIntegrationTest
public class AppointmentRelatedEndpointsIT {

    FhirR4Util fhirR4Util = new FhirR4Util();

    @TestHTTPResource("/fhir/r4")
    String url;

    @Test
    public void schedule1_fetch_all_defined_areas() {
        System.out.println(url);
        Appointment appointment = fhirR4Util.getGenericClient(url).read().resource(Appointment.class)
                .withId("D3eeb6d9398e45029093231").execute();
        assertNotNull(appointment);
    }
```

If I set `@TestHTTPResource(value = "/fhir/r4", tls = true)` it is however resolved against `https`!

**Which pages**
(add links or names for the affected pages)

I just did the demo application (generated using `quarkus create app ch.demo:47582 --maven --no-wrapper --java=21`), and modified 
```
@QuarkusTest
class GreetingResourceTest {

    @TestHTTPResource("/")
    String url;

    @Test
    void testHelloEndpoint() {
        System.out.println("URL: " + url);
        given()
          .when().get("/hello")
          .then()
             .statusCode(200)
             .body(is("Hello from Quarkus REST"));
    }

}
```
calling:
```
mvn verify -Dquarkus.http.test-ssl-enabled=true -Dquarkus.http.test-host=somehost.info -Dquarkus.http.test-ssl-port=443 -DskipITs=false
```
outputs 
```
[INFO] Running ch.demo.GreetingResourceIT
URL: http://somehost.info:8081/
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.428 s <<< FAILURE! -- in ch.demo.GreetingResourceIT
```

--- a/docs/src/main/asciidoc/getting-started-testing.adoc
+++ b/docs/src/main/asciidoc/getting-started-testing.adoc
@@ -1431,7 +1431,7 @@ An example use of this could be the following Maven command, that forces `@Quark
 ./mvnw verify -Dquarkus.http.test-host=1.2.3.4 -Dquarkus.http.test-port=4321
 ----
 
-To test against a running instance that only accepts SSL/TLS connection (example: `https://1.2.3.4:4321`) set the system property `quarkus.http.test-ssl-enabled` to `true`.
+To test against a running instance that only accepts SSL/TLS connection (example: `https://1.2.3.4:4321`) set the system property `quarkus.http.test-ssl-enabled` to `true` and `quarkus.http.test-ssl-port` to the target HTTPS port.
 
 == Mixing `@QuarkusTest` with other type of tests
 

--- a/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResource.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResource.java
@@ -26,7 +26,7 @@ public @interface TestHTTPResource {
      * @return If the URL should use the HTTPS protocol and SSL port
      * @deprecated use #tls instead
      */
-    @Deprecated
+    @Deprecated(since = "3.10", forRemoval = true)
     boolean ssl() default false;
 
     /**

--- a/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java
@@ -90,7 +90,8 @@ public class TestHTTPResourceManager {
                         path = endpointPath;
                     }
                     String val;
-                    if (resource.ssl() || resource.tls()) {
+                    if (resource.ssl() || resource.tls() || ConfigProvider.getConfig()
+                            .getOptionalValue("quarkus.http.test-ssl-enabled", Boolean.class).orElse(false)) {
                         if (management) {
                             if (path.startsWith("/")) {
                                 val = getManagementSslUri() + path;

